starlette/concurrency.py

```diff
@@ -50,8 +50,9 @@ def _next(iterator: typing.Iterator[T]) -> T:
 
 
 async def iterate_in_threadpool(
-    iterator: typing.Iterator[T],
-) -> typing.AsyncIterator[T]:
+    iterable: typing.Iterable[T],
+) -> typing.AsyncIterable[T]:
+    iterator = iter(iterable)
```

Comment:
Can you create a test case that fails without this?

Reply:
Sure, I can cook something up 👍

Reply:
This test fails on `master` but passes here

```python
@pytest.mark.anyio
async def test_iterate_in_threadpool() -> None:
    class CustomIterable:
        def __iter__(self):
            yield from range(3)

    assert [v async for v in iterate_in_threadpool(CustomIterable())] == [0, 1, 2]
```

Let me know if you were hoping for something different